app/assets/javascripts/components/articles/add_available_articles.jsx

```diff
@@ -64,17 +65,26 @@ const AddAvailableArticles = createReactClass({
   },
 
   render() {
+    const isWikidata = this.props.project === 'wikidata';
     return (
       <div className="pop__padded-content">
-        <TextAreaInput
+        {isWikidata && <TextAreaInput
```

Comment:
Is there a simpler way to do this, such as defining the placeholder before the `return`?

Reply:
I tried this:
![placeholder](https://user-images.githubusercontent.com/65791349/147284898-fac1046c-92fb-4428-ae9e-b08656e19013.png)

No good. Like a few other things, I could think of. This ugly stuff was the only thing that works for endless switching from wiki to data and reverse

Reply:
Hmm... I'm not sure why that wouldn't work. The component obviously re-renders when the project changes, or else neither implementation would work.

I'll play around with it when I have a chance. This seems like it might be related to rendering shortcuts with TextAreaInput.

Reply:
With previous code (and all others) everything is changing nicely, but the placeholder. It is switching only from wiki to data and only one time.

Reply:
The simpler way works if you also change the `id` to correspond to the current state of `isWikidata`. That forces React to use two different DOM objects (and component instances) for the two versions of this `TextAreaInput` component, so it works around whatever is preventing it from re-rendering when the `placeholder` value changes.

I think the reason it was not re-rendering is that the TextAreaInput component is using the `InputHOC` higher-order-component, and that component implements a custom `shouldComponentUpdate` implementation that limits re-renders to `state` (not `prop`) changes in order to keep the input field performance fast enough that you don't wait for a character to render when you type it.

Reply:
(Here's what I did for the id: `const inputId = isWikidata ? 'add_available_items' : 'add_available_articles';`)

Reply:
Works great with the id. Thanks for the explanation, there is a lot things I didn't know